### DISCUSSION TITLE
Why is my subgraph not connected?

### DISCUSSION THREAD

**Discussion by asyouknowme** [inducer/pymetis#37] [Troubleshooting]
Hi, 
  Through graph partition, The connectivity of the subgraph I get is not good. There are 1 million nodes in the graph. Use the following code to determine, some subgraphs are even 8.
`nx.number_connected_components(subgraph)`

**Comment by inducer** [comment]
> not good

That's not very precise.

- Can you supply code to reproduce the issue?
- What were you expecting? What happened instead?

Relatedly, is the behavior you see different from metis? If yes, it's an issue with pymetis. If not, it's an issue with metis itself.

I'll convert this to a troubleshooting question for now.

**Reply by asyouknowme** [reply]
http://snap.stanford.edu/data/soc-Pokec.html
I use the edgelist in this dataset. Adjacency is built through the neighbors of nodes. 

```
import numpy as np
import pymetis
import pandas as pd
import networkx as nx

edges = pd.read_csv("./soc-pokec-relationships/soc-pokec-relationships.txt",header=None,sep='\t')
edges = edges.applymap(lambda x:x-1)
G = nx.Graph()
G.add_edges_from(edges.to_numpy())

adj_l = []
num_nodes = list(G.nodes)
num_nodes.sort()
for i in num_nodes:
    adj_l.append(np.array(list(G.neighbors(i))))

n = G.number_of_nodes() // 200
n_cuts, membership = pymetis.part_graph(n,adjacency=adj_l)
clusters = list(set(membership))
cluster_membership = {node: membership for node, membership in enumerate(membership)}
for cluster in clusters:
            sub_graph = G.subgraph(
                [node for node in sorted(G.nodes()) if cluster_membership[node] == cluster])
            print(nx.number_connected_components(sub_graph))
```
Every node in a subgraph should be able to reach the rest of the nodes in that subgraph. in other words, the final output should be 1. but obviously not all 1. I haven't tried metis.

**Reply by asyouknowme** [reply]
Sorry, I think it might be a problem with the data, it might not be well partitioned.
